src/test/java/org/opensearch/security/tools/democonfig/CertificateGeneratorTests.java

```diff
@@ -0,0 +1,155 @@
+import static org.hamcrest.Matchers.is;
+import static org.opensearch.security.tools.democonfig.util.DemoConfigHelperUtil.createDirectory;
+import static org.opensearch.security.tools.democonfig.util.DemoConfigHelperUtil.deleteDirectoryRecursive;
+import static org.junit.Assert.fail;
+
+public class CertificateGeneratorTests {
+
+    private static Installer installer;
+
+    @Before
+    public void setUp() {
+        installer = Installer.getInstance();
+        installer.buildOptions();
+        installer.OPENSEARCH_CONF_DIR = System.getProperty("user.dir") + File.separator + "test-conf";
+        createDirectory(installer.OPENSEARCH_CONF_DIR);
+    }
+
+    @After
+    public void tearDown() {
+        deleteDirectoryRecursive(installer.OPENSEARCH_CONF_DIR);
+        Installer.resetInstance();
+    }
+
+    @Test
+    public void testCreateDemoCertificates() {
+        CertificateGenerator certificateGenerator = new CertificateGenerator(installer);
+        Certificates[] certificatesArray = Certificates.values();
+
+        certificateGenerator.createDemoCertificates();
+
+        for (Certificates cert : certificatesArray) {
+            String certFilePath = installer.OPENSEARCH_CONF_DIR + File.separator + cert.getFileName();
+            File certFile = new File(certFilePath);
+            assertThat(certFile.exists(), is(equalTo(true)));
+            assertThat(certFile.canRead(), is(equalTo(true)));
+
+            if (certFilePath.endsWith("-key.pem")) {
+                checkPrivateKeyValidity(certFilePath);
+                return;
+            }
+            checkCertificateValidity(certFilePath);
+        }
+    }
+
+    @Test
+    public void testCreateDemoCertificates_invalidPath() {
+        installer.OPENSEARCH_CONF_DIR = "invalidPath";
+        CertificateGenerator certificateGenerator = new CertificateGenerator(installer);
+        try {
+            System.setSecurityManager(new NoExitSecurityManager());
+            certificateGenerator.createDemoCertificates();
+        } catch (SecurityException e) {
+            assertThat(e.getMessage(), equalTo("System.exit(-1) blocked to allow print statement testing."));
+        } finally {
+            System.setSecurityManager(null);
+        }
+    }
+
+    private static void checkCertificateValidity(String certPath) {
+        try (FileInputStream certInputStream = new FileInputStream(certPath)) {
+            CertificateFactory cf = CertificateFactory.getInstance("X.509");
+            Certificate certificate = cf.generateCertificate(certInputStream);
+
+            if (certificate instanceof X509Certificate) {
+                X509Certificate x509Certificate = (X509Certificate) certificate;
+                x509Certificate.checkValidity();
+
+                Date expiryDate = x509Certificate.getNotAfter();
+                Instant expiry = expiryDate.toInstant();
+
+                assertThat(isExpiryAtLeastAYearLater(expiry), is(equalTo(true)));
```

Comment:
nit: you can use `org.hamcrest.Matchers.greaterThan` to verify the time, then you'll get a meaningful error if the check fails.  Checking `equalTo(true)` doesn't produce a useful output

Reply:
agreed and done.